src/app/shared/couchdb.service.ts

```diff
@@ -61,7 +61,19 @@ export class CouchService {
             .then(this.handleRes)
             .catch(this.handleError);
     }
-    
+
+    //For Attachment files
+    saveAttachment(db:string,data:any,content_type:any): Promise<any> {

```

Comment:
Ah, I see the slight difference with the `content_type` parameter.  Rather than write a completely new function, the opts parameter for the `put` method can be used to overwrite the header.